src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -1273,10 +1273,68 @@ public void GenerateVTable(Class @class)
 
             GenerateVTableClassSetup(@class, wrappedEntries);
 
+            GenerateVTableCleanup(@class, wrappedEntries);
+
             WriteLine("#endregion");
             PopBlock(NewLineKind.BeforeNextBlock);
         }
 
+        private void GenerateVTableCleanup(Class @class, IList<VTableComponent> wrappedEntries)
+        {
+            WriteLine("private static readonly Destructor finalise = new Destructor();");
+            WriteLine("private sealed class Destructor");
+            WriteStartBraceIndent();
+            WriteLine("~Destructor()");
+            WriteStartBraceIndent();
+            DeAllocateVTables(@class, wrappedEntries, false);
+            WriteCloseBraceIndent();
+            WriteCloseBraceIndent();
+            NewLine();
+        }
+
+        private void DeAllocateVTables(Class @class, IList<VTableComponent> wrappedEntries,
+            bool destructorOnly)
+        {
+            if (Options.IsMicrosoftAbi)
+                DeAllocateVTablesMS(@class, wrappedEntries, destructorOnly);
+            else
+                DeAllocateVTablesItanium(@class, wrappedEntries, destructorOnly);
+        }
+
+        private void DeAllocateVTablesMS(Class @class, IList<VTableComponent> wrappedEntries,
+            bool destructorOnly)
+        {
+            if (destructorOnly)
```

Comment:
Actually ```__ManagedVTablesDtorOnly``` is not defined in some cases. That's why I did this. But, now I've changed it.